test/providers/summary/scancode-new.js

```diff
@@ -0,0 +1,205 @@
+    }
+  })
+
+  it('summarizes large npm', () => {
+    const coordinates = { type: 'npm', provider: 'npmjs' }
+
+    for (let version of scancodeVersions) {
+      const harvestData = getHarvestData(version, 'npm-large')
+      const result = summarizer.summarize(coordinates, harvestData)
+      assert.equal(
+        result.licensed.declared,
+        'BSD-3-Clause OR GPL-2.0',
+        `Declared license mismatch for version ${version}`
+      )
+      assert.equal(result.described.releaseDate, '2018-03-31', `releaseDate mismatch for version ${version}`)
+      assert.deepEqual(uniq(flatten(result.files.map((x) => x.attributions))).filter((x) => x).length, 33)
+      assert.deepEqual(result.files.find((x) => x.path === 'package/LICENSE').natures, ['license'])
+      assert.equal(flatten(result.files.map((x) => x.natures)).filter((x) => x).length, 1)
+    }
+  })
+
+  it('summarizes maven with a complex declared license', () => {
+    const coordinates = { type: 'maven', provider: 'mavencentral' }
+    const harvestData = getHarvestData('32.0.8', 'maven-complex-declared-license')
+    const result = summarizer.summarize(coordinates, harvestData)
+
+    assert.equal(result.licensed.declared, 'BSD-3-Clause-No-Nuclear-Warranty AND MIT')
+  })
+
+  it('summarizes github with a single declared license', () => {
+    const coordinates = { type: 'git', provider: 'github' }
+    const harvestData = getHarvestData('32.0.8', 'github-single-declared-license')
+    const result = summarizer.summarize(coordinates, harvestData)
+
+    assert.equal(result.licensed.declared, 'MIT')
+  })
+
+  it('summarizes pypi with a complex declared license', () => {
+    const coordinates = { type: 'pypi', provider: 'pypi' }
+    const harvestData = getHarvestData('32.0.8', 'pypi-complex-declared-license')
+    const result = summarizer.summarize(coordinates, harvestData)
+    assert.equal(result.licensed.declared, 'HPND')
+  })
+
+  it('summarizes github with a single declared license', () => {
+    const coordinates = { type: 'git', provider: 'github' }
+    const harvestData = getHarvestData('32.0.8', 'github-single-declared-license')
+    const result = summarizer.summarize(coordinates, harvestData)
+
+    assert.equal(result.licensed.declared, 'MIT')
+  })
+
+  it('should detect license from maven license file', () => {
+    const coordinates = { type: 'maven', provider: 'mavencentral' }
+    const harvestData = getHarvestData('32.0.8', 'maven-flywaydb-file-license')
+    const result = summarizer.summarize(coordinates, harvestData)
+    assert.equal(result.licensed.declared, 'Apache-2.0')
+  })
+
+  it('summarizes using license_expression', () => {
+    const coordinates = { type: 'debsrc', provider: 'debian' }
+    const harvestData = getHarvestData('32.0.8', 'debsrc-license-expression')
+    const result = summarizer.summarize(coordinates, harvestData)
+    assert.equal(result.licensed.declared, 'Apache-2.0')
+  })
```

Comment:
This test currently fails due to https://github.com/nexB/scancode-toolkit/issues/3690

In the meantime, I'm wondering if we could build a usable workaround out of the new data at https://github.com/clearlydefined/service/blob/dac4efb9f62380da59e69af3226bc91b59a97326/test/fixtures/scancode/32.0.8/debsrc-license-expression.json#L85-L98 ? 🤔

Reply:
If we decide to skip this specific test for the moment, I'd love suggestions for other test cases I could add that fall back to the `packages[0].license_expression`

Reply:
More test cases can be found in [commit message] (https://github.com/clearlydefined/service/pull/1018/commits/57c59011743df9c6e329255c729f97ed9cdf59dd) for the fix.

Reply:
> In the meantime, I'm wondering if we could build a usable workaround out of the new data at

The other license expressions seems to align with discovered license, but may not align with declared.
Failing to detect license at declared license expression, and also at package[0].  The next route to try is to find the license file at the package root level or a specific license file folder.  In v3 and v30, file.is_license_text is the flag to indicate a license file.  This flag seems to be removed? Is there a new way to indicate license file in v32?
There are two files with "is_legal": true, but none of them at the package root level:
- "path": "tenacity-8.2.1/LICENSE"
- "path": "debian/copyright"
Is there a convention to organize debian packages and specification on a designated folder to put license file?  If so, can potentially go to that specific folder to look for license file.

Reply:
> file.is_license_text is the flag to indicate a license file. This flag seems to be removed?

Indeed: 

> The deprecated "--is-license-text" option has been removed. This is now built-in with the --license-text option and --info and exposed with the "percentage_of_license_text" attribute.

https://github.com/nexB/scancode-toolkit/blob/develop/CHANGELOG.rst#license-detection

> Is there a new way to indicate license file in v32?

Yes, I've made updates to the function now called `_getLicenseFromLicenseDetections` to use `percentage_of_license_text >= 80`. The problem was that `_getDetectedLicenseFromFiles` was not working for `debsrc` components because the component name does not match the actual package folder name, `python-tenacity` vs `tenacity-8.2.1`. I've pushed a commit with a suggestion for how this could be supported: https://github.com/clearlydefined/service/pull/1056/commits/49c3b470415ffe9b851969ef5225334374834969

Additionally, I was wondering if we don't want to maybe look at more than `package[0]` and maybe iterate over all packages? The `tenacity` package element in the `packages` array has all the right info:

https://github.com/clearlydefined/service/blob/8bc66a722b16cd84c390a129bd5c1dc371cf8bab/test/fixtures/scancode/32.0.8/debsrc-license-expression.json#L297-L341

Reply:
Thanks for filing all the issues to ScanCode!
> I've made updates to the function now called `_getLicenseFromLicenseDetections` to use `percentage_of_license_text >= 80`. 

I believe that determining the limit (e.g. 80) should be a decision made by the community and properly documented, as mentioned in [Jeff's comment in another PR](https://github.com/clearlydefined/service/pull/999#discussion_r1353485659). To provide some context, in our handling of scancode data prior to version 3.0.0, licenses with a `score` of 80 or higher were considered file-level licenses, whereas licenses with a `score` of 90 or higher were considered declared licenses for the package. For more information, please refer to the [discussion thread](https://github.com/clearlydefined/service/pull/999#discussion_r1353485659). It appears that there is a distinction between identifying a license file and a declared license for the package.

> Additionally, I was wondering if we don't want to maybe look at more than `package[0]` and maybe iterate over all packages? The `tenacity` package element in the `packages` array has all the right info:
> 

The package[2].type is pypi and it points to tenacity 8.2.1, which has a "declared_license_expression" of "apache-2.0". The pypi package [pypi/pypi/-/tenacity/8.2.1](https://clearlydefined.io/definitions/pypi/pypi/-/tenacity/8.2.1) is declared as Apache-2.0.

The question then arises: is it possible for a package (in this case, pypi/pypi/-/tenacity/8.2.1) to be repackaged and published (as debsrc/debian/-/python-tenacity/8.2.1-1) under a different license? @yashkohli88 @capfei, could you please provide some feedback on this?

Reply:
> Is there a convention to organize debian packages and specification on a designated folder to put license file? If so, can potentially go to that specific folder to look for license file.

Looks like that debian/copyright is the required license info.  See [Debian New Maintainers' Guide](https://www.debian.org/doc/manuals/maint-guide/dreq.en.html)

Reply:
> Looks like that debian/copyright is the required license info.

Unfortunately, `debian/copyright` appears to contain a lot of non-license text as well, so its `percentage_of_license_text` is just over 53% 🤔

For the rest of this logic, I should now be mirroring the logic from the existing ScanCode summarizer and the logic from the old `--is-license-text` option as it was originally implemented: https://github.com/nexB/scancode-toolkit/blob/v30.1.0/src/licensedcode/plugin_license_text.py`